Common/Data/Custom/Intrinio/IntrinioEconomicData.cs

```diff
@@ -0,0 +1,256 @@
+                    // In backtesting mode, we need the data in ascending order...
+                    order = "asc";
+                    // and avoid making furthers calls for the same data.
+                    _firstTime = false;
+                }
+
+                var item = GetStringForDataTransformation(_dataTransformation);
+                var url = $"{_baseUrl}identifier={config.Symbol.Value}&item={item}&sort_order={order}";
+                var byteKey = Encoding.ASCII.GetBytes($"{IntrinioConfig.User}:{IntrinioConfig.Password}");
+                var authorizationHeaders = new List<KeyValuePair<string, string>>
+                {
+                    new KeyValuePair<string, string>("Authorization",
+                                                     $"Basic ({Convert.ToBase64String(byteKey)})")
+                };
+
+                subscription = new SubscriptionDataSource(url, SubscriptionTransportMedium.RemoteFile, FileFormat.Csv,
+                                                          authorizationHeaders);
+                LastIntrinoAPICall = DateTime.Now;
+            }
+            else
+            {
+                subscription = new SubscriptionDataSource("", SubscriptionTransportMedium.LocalFile);
+            }
+            return subscription;
+        }
+
+        /// <summary>
+        ///     Reader converts each line of the data source into BaseData objects. Each data type creates its own factory method,
+        ///     and returns a new instance of the object
+        ///     each time it is called. The returned object is assumed to be time stamped in the config.ExchangeTimeZone.
+        /// </summary>
+        /// <param name="config">Subscription data config setup object</param>
+        /// <param name="line">Line of the source document</param>
+        /// <param name="date">Date of the requested data</param>
+        /// <param name="isLiveMode">true if we're in live mode, false for backtesting mode</param>
+        /// <returns>
+        ///     Instance of the T:BaseData object generated by this line of the CSV
+        /// </returns>
+        public override BaseData Reader(SubscriptionDataConfig config, string line, DateTime date, bool isLiveMode)
+        {
+            var obs = line.Split(',');
+            var time = DateTime.MinValue;
+            if (!DateTime.TryParseExact(obs[0], "yyyy-MM-dd", CultureInfo.InvariantCulture, DateTimeStyles.None,
+                                        out time)) return null;
+            var value = obs[1].ToDecimal();
+            return new IntrinioEconomicData
+            {
+                Symbol = config.Symbol,
+                Time = time,
+                EndTime = time + QuantConnect.Time.OneDay,
+                Value = value
+            };
+        }
+
+        /// <summary>
+        /// Set the Intrinio API user and password.
+        /// </summary>
+        public static void SetUserAndPassword(string user, string password)
```

Comment:
Out of curiosity, why place this set method here instead of on `IntrinioConfig` ?

If this were placed on `IntrinioConfig` then the setters could be made private which is kind of nice, forces people to invoke he set username/password method directly.